lib/src/handler_client.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'base_client.dart';
+import 'request.dart';
+import 'response.dart';
+
+typedef Future<Response> Handler(Request request);
```

Comment:
I'd put this in its own library. I know it seems silly when it's just one line of code, but having one file per type makes them a lot easier to track down in the source.